src/ReactiveUI/Suspension/SuspensionHostExtensions.cs

```diff
@@ -16,38 +17,53 @@ namespace ReactiveUI;
 public static class SuspensionHostExtensions
 {
     /// <summary>
-    /// Observe changes to the AppState of a class derived from ISuspensionHost.
+    /// Lock to ensure app state is loaded only once.
     /// </summary>
-    /// <typeparam name="T">The observable type.</typeparam>
+    private static readonly object appStateInitializationLock = new object();
+
+    /// <summary>
+    /// Supsension driver reference field to prevent introducing breaking change.
+    /// </summary>
+    private static ISuspensionDriver? suspensionDriver;
+
+    /// <summary>
+    /// Get the current App State of a class derived from ISuspensionHost.
+    /// </summary>
+    /// <typeparam name="T">The app state type.</typeparam>
     /// <param name="item">The suspension host.</param>
-    /// <returns>An observable of the app state.</returns>
-    public static IObservable<T> ObserveAppState<T>(this ISuspensionHost item)
-        where T : class
+    /// <returns>The app state.</returns>
+    public static T GetAppState<T>(this ISuspensionHost item)
     {
         if (item is null)
         {
             throw new ArgumentNullException(nameof(item));
         }
 
-        return item.WhenAny(suspensionHost => suspensionHost.AppState, observedChange => observedChange.Value)
-                   .WhereNotNull()
-                   .Cast<T>();
+        if (item.AppState is null)
+        {
+            item.EnsureLoadAppState();
```

Comment:
This would be a threading issue. If you use the interlock approach and always call the method could avoid that